src/MCPClient/lib/clientScripts/handleUnstoredDIP.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+
+# This file is part of Archivematica.
+#
+# Copyright 2010-2017 Artefactual Systems Inc. <http://artefactual.com>
+#
+# Archivematica is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Archivematica is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Archivematica.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Handle Unstored DIP.
+
+This client script hanles an unstored DIP by moving it to the rejected/
+directory if it has *not* been uploaded and doing nothing otherwise.
+"""
+from __future__ import print_function
+
+import os
+import shutil
+import sys
+
+import django
+django.setup()
+# archivematicaCommon
+from custom_handlers import get_script_logger
+
+
+logger = get_script_logger('archivematica.mcp.client.handleUnstoredDIP')
+
+
+def _get_sip_dirname(sip_path):
+    return os.path.basename(sip_path.rstrip('/'))
+
+
+def _sip_uploaded(sip_path, uploaded_dips_path):
+    """Return ``True`` if the SIP has been uploaded."""
+    return _get_sip_dirname(sip_path) in os.listdir(uploaded_dips_path)
```

Comment:
Is there any reason why you can't just use `os.path.isdir(path)` or `os.path.exists(path)`? Sounds like it'd be more efficient because it just uses `stat()`.